src/axolotl/utils/data.py

```diff
@@ -905,3 +906,30 @@ def encode_packed_pretraining(
             chunked_data[feature].append(collated_features[feature].squeeze(0))
 
     return chunked_data
+
+
+def load_prepare_dpo_datasets(cfg):
+    def load_split(dataset_cfgs, _cfg):
+        split_datasets: List[Any] = []
+        for i, ds_cfg in enumerate(dataset_cfgs):
+            ds = load_dataset(  # pylint: disable=invalid-name
+                ds_cfg["path"], split=ds_cfg["split"]
+            )
+            split_datasets.insert(i, ds)
```

Comment:
```suggestion
        for i, ds_cfg in enumerate(dataset_cfgs):
            if ds_cfg["ds_type"] == "json":
                for data_file in ds_cfg["data_files"]:
                    data_files = {ds_cfg["split"]: data_file}
                    ds = load_dataset("json", data_files=data_files, split=ds_cfg["split"])
                    split_datasets.insert(i, ds)
            else:
                ds = load_dataset(  # pylint: disable=invalid-name
                    ds_cfg["path"], split=ds_cfg["split"]
                )
                split_datasets.insert(i, ds)
```

Reply:
This change would allow for local datasets as mentioned in https://github.com/OpenAccess-AI-Collective/axolotl/pull/1126#discussion_r1453942204

Reply:
thanks!